falcon/util/uri.py

```diff
@@ -376,3 +376,31 @@ def parse_host(host, default_port=None):
     # or a domain name plus a port
     name, _, port = host.partition(':')
     return (name, int(port))
+
+
+def unquote_string(quoted):
+    """unquote a rfc7320 "quoted-string" into normal one
+
+    Args:
+        quoted (str): Original quoted string
+
+    Returns:
+        str: unquoted string
+
+    Raises:
+        TypeError: `quoted` was not a ``str``.
+    """
+    tmp_quoted = quoted.strip()
+    if len(tmp_quoted) < 2:
+        return quoted
+    elif tmp_quoted[0] != '"' or tmp_quoted[-1] != '"':
+        # return original one, prevent side-effect
+        return quoted
+    tmp_quoted = tmp_quoted[1:-1]
+    if '\\' not in tmp_quoted:
```

Comment:
Would you mind prefacing L400-406 with a comment explaining why this approach was taken? E.g.,:

```python
# PERF(philiptzou): blah blah blah...
```